From f5180847e0fb848d84a8be0966cd21369c26fe32 Mon Sep 17 00:00:00 2001
From: Pedro Palazon Candel <pedro@joyent.com>
Date: Tue, 28 Feb 2017 18:47:05 +0100
Subject: [PATCH] TOOLS-1704 `sdcadm post-setup cmon` should require that CNS
 be setup Reviewed by: Trent Mick <trent.mick@joyent.com>

---
 CHANGES.md             |  5 +++++
 lib/post-setup/cmon.js | 18 ++++++++++++++++++
 package.json           |  2 +-
 3 files changed, 24 insertions(+), 1 deletion(-)

diff --git a/CHANGES.md b/CHANGES.md
index 278ea71..8b96829 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -10,6 +10,11 @@
 
 # sdcadm Changelog
 
+
+## 1.15.6
+
+- TOOLS-1704 'sdcadm post-setup cmon' requires CNS being setup
+
 ## 1.15.5
 
 - TOOLS-1651 sdcadm create should support agent instances.
diff --git a/lib/post-setup/cmon.js b/lib/post-setup/cmon.js
index a4aecaf..5e90117 100644
--- a/lib/post-setup/cmon.js
+++ b/lib/post-setup/cmon.js
@@ -91,6 +91,24 @@ function do_cmon(subcmd, opts, args, cb) {
             });
         },
 
+        function ensureCnsSvc(ctx, next) {
+            self.sdcadm.sapi.listServices({
+                name: 'cns',
+                application_uuid: self.sdcadm.sdc.uuid
+            }, function (svcErr, svcs) {
+                if (svcErr) {
+                    next(svcErr);
+                } else if (!svcs.length) {
+                    next(new errors.UpdateError(
+                        'The CNS service is required by CMON.\n' +
+                        common.indent('Please, install it with ' +
+                            '`sdcadm post-setup cns`.')));
+                } else {
+                    next();
+                }
+            });
+        },
+
         /* @field ctx.cmonPkg */
         function getPkg(ctx, next) {
             var filter = {name: svcData.params.package_name,
diff --git a/package.json b/package.json
index 86858d7..8214f3f 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "sdcadm",
   "description": "Administer a SmartDataCenter (SDC) standup",
-  "version": "1.15.5",
+  "version": "1.15.6",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
-- 
2.21.0

